src/foundation.cs

```diff
@@ -3489,8 +4253,58 @@ interface NSLinguisticTagger {
 		[NullAllowed] // by default this property is null
 		[Export ("string", ArgumentSemantic.Retain)]
 		string AnalysisString { get; set; }
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("tagsInRange:unit:scheme:options:tokenRanges:")]
+		string[] GetTags (NSRange range, NSLinguisticTaggerUnit unit, string scheme, NSLinguisticTaggerOptions options, [NullAllowed] out NSValue[] tokenRanges);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("enumerateTagsInRange:unit:scheme:options:usingBlock:")]
+		void EnumerateTags (NSRange range, NSLinguisticTaggerUnit unit, string scheme, NSLinguisticTaggerOptions options, LinguisticTaggerEnumerateTagsBlock block);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("tagAtIndex:unit:scheme:tokenRange:")]
+		[return: NullAllowed]
+		string GetTag (nuint charIndex, NSLinguisticTaggerUnit unit, string scheme, [NullAllowed] NSRange tokenRange);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Export ("tokenRangeAtIndex:unit:")]
+		NSRange GetTokenRange (nuint charIndex, NSLinguisticTaggerUnit unit);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Static]
+		[Export ("availableTagSchemesForUnit:language:")]
+		string[] GetAvailableTagSchemes (NSLinguisticTaggerUnit unit, string language);
+
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[NullAllowed, Export ("dominantLanguage")]
+		string DominantLanguage { get; }
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Static]
+		[Export ("dominantLanguageForString:")]
+		[return: NullAllowed]
+		string GetDominantLanguage (string str);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Static]
+		[Export ("tagForString:atIndex:unit:scheme:orthography:tokenRange:")]
+		[return: NullAllowed]
+		string GetTag (string str, nuint charIndex, NSLinguisticTaggerUnit unit, string scheme, [NullAllowed] NSOrthography orthography, [NullAllowed] NSRange tokenRange);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Static]
+		[Export ("tagsForString:range:unit:scheme:options:orthography:tokenRanges:")]
+		string[] GetTags (string str, NSRange range, NSLinguisticTaggerUnit unit, string scheme, NSLinguisticTaggerOptions options, [NullAllowed] NSOrthography orthography, [NullAllowed] out NSValue[] tokenRanges);
+
+		[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+		[Static]
+		[Export ("enumerateTagsForString:range:unit:scheme:options:orthography:usingBlock:")]
+		void EnumerateTagsForString (string str, NSRange range, NSLinguisticTaggerUnit unit, string scheme, NSLinguisticTaggerOptions options, [NullAllowed] NSOrthography orthography, LinguisticTaggerEnumerateTagsBlock block);
 	}
 
+	delegate void LinguisticTaggerEnumerateTagsBlock (string tag, NSRange tokenRange, bool stop);
```

Comment:
`Block` sounds strange in C#, maybe use `Callback`, or something even more different, like `LinguisticTagEnumerator`.